sympy/solvers/ode/ode.py

```diff
@@ -7281,14 +7284,12 @@ def _linear_2eq_order2_type11(x, y, t, r, eq):
 
 def sysode_linear_neq_order1(match):
     from sympy.solvers.ode.systems import (_linear_neq_order1_type1,
-        _linear_neq_order1_type3, _linear_neq_order1_type2)
-
-    if match['type_of_equation'] == 'type1':
-        sol = _linear_neq_order1_type1(match)
-    elif match['type_of_equation'] == 'type2':
-        sol = _linear_neq_order1_type2(match)
-    elif match['type_of_equation'] == 'type3':
-        sol = _linear_neq_order1_type3(match)
+        _linear_neq_order1_type2, _linear_neq_order1_type3,
+        _linear_neq_order1_type4)
+
+    solver = locals()['_linear_neq_order1_{}'.format(match['type_of_equation'])]
+    sol = solver(match)
```

Comment:
Using `locals` like this is a bad idea. We should not do this and should also remove the other places where this happens

Reply:
Other places, they are using `globals` instead of locals, do they have to be changed to?